src/key.js

```diff
@@ -1628,7 +1637,8 @@ export async function getPreferredAlgo(type, keys, date=new Date(), userId={}) {
   const prioMap = {};
   await Promise.all(keys.map(async function(key) {
     const primaryUser = await key.getPrimaryUser(date, userId);
-    if (!primaryUser || !primaryUser.selfCertification[prefProperty]) {
+    const noSelfCertOk = !config.require_uid_self_cert && primaryUser && !primaryUser.selfCertification;
+    if (!primaryUser || noSelfCertOk || !primaryUser.selfCertification[prefProperty]) {
```

Comment:
I didn't want to - I thought it would be safer if all functions on their own still behaved just like before `require_uid_self_cert` was available. That way there is less room for error / less potential unexpected behavior in the future.

If you strongly prefer that approach, I'll change it. Let me know.

Reply:
I agree with you on principle, but in this particular case the logic is basically "if we cannot extract an algorithm preference from the primary user, return the default." For the purposes of this function, the only thing your config option changes is that we can no longer guarantee that a selfCertification is present, so I see the addition of the `!primaryUser.selfCertification` check as just making sure that the previously existing `primaryUser.selfCertification[prefProperty]` check does not throw an error. For that reason I personally don't see any room for error here, but if you do let me know :)